plasmapy/particles/exceptions.py

```diff
@@ -1,41 +1,61 @@
 """
-Collection of `Exceptions` and `Warnings` for PlasmaPy particles.
+Collection of exceptions and warnings for `plasmapy.particles`.
 """
 __all__ = [
-    "AtomicError",
-    "AtomicWarning",
+    "ParticleError",
+    "ParticleTypeError",
+    "ParticleValueError",
+    "ParticleWarning",
     "ChargeError",
     "InvalidElementError",
     "InvalidIonError",
     "InvalidIsotopeError",
     "InvalidParticleError",
-    "MissingAtomicDataError",
-    "MissingAtomicDataWarning",
+    "MissingParticleDataError",
+    "MissingParticleDataWarning",
     "UnexpectedParticleError",
 ]
 
 from plasmapy.utils import PlasmaPyError, PlasmaPyWarning
 
 
-class AtomicError(PlasmaPyError):
-    """An exception for errors in the `~plasmapy.particles` subpackage."""
+class ParticleError(PlasmaPyError):
+    """Base exception for errors in the `~plasmapy.particles` subpackage."""
 
     pass
 
 
-class MissingAtomicDataError(AtomicError):
+class ParticleTypeError(ParticleError, TypeError):
+    """
+    An exception for errors in the `~plasmapy.particles` subpackage
+    caused because an incorrect type was provided.
+    """
+
+    pass
+
+
+class ParticleValueError(ParticleError, ValueError):
+    """
+    An exception for errors in the `~plasmapy.particles` subpackage
+    caused because an incorrect value was provided.
+    """
+
+    pass
+
+
+class MissingParticleDataError(ParticleError):
     """An exception for missing atomic or particle data."""
 
     pass
 
 
-class ChargeError(AtomicError):
+class ChargeError(ParticleError):
```

Comment:
```suggestion
class ParticleChargeError(ParticleError):
```

To make error reporting more explicit.  Without context, it might not be immediately apparent the `ChargeError` is really referring to a `ParticleChargeError`.

Reply:
`ChargeError` is a bit broader than just for particles, since it could refer to missing or invalid charge information elsewhere.  I'm wondering if it'd be better to put this in `plasmapy.utils.exceptions`...but that's not for this pull request.

Reply:
I'm in two minds about this.  I'm afraid we're heading down the path of having things like `MassError`, `VelocityError`, `DensityError`, etc.  Something like that would end up being a pain to maintain, especially since most cases would be better suited using `ValueError` or `TypeError`.  Now, I don't mind it being used in the context of `plasmapy.particles` to indicate this is an error is coming from the particle framework.  In that case, it should only be used within the particle framework.  If it would be used in a more general sense, then I take issue according to my initial point but it should also not inherit for `ParticleError`.  As stated in the `ParticleError` docstring, `ParticleError` is for errors specific to the particle framework.

Reply:
`ChargeError` is a bit different than `MassError`, etc.  A `ChargeError` can come up when a function requires a charged particle, but the particle is not charged or lacks charge information, or when an invalid charge is supplied (i.e. H²⁺).  I've found it useful in the `particles` subpackage.  I'd rather leave the status quo for `ChargeError` at least for the time being, in part to avoid scope creep for this PR.